docs/GETTING_STARTED.md

```diff
@@ -0,0 +1,239 @@
+
+```
+sudo ./soci push --plain-http localhost:5000/rabbitmq:latest
+```
+
+This will push all of the ztoc files and the soci index manifest.
+
+## Running the image
+
+### Configuring containerd
+To have containerd use the soci-snapshotter we need to reconfigure it.  
+Containerd is a daemon managed using the systemd / systemctl tools.  
+First let's stop containerd
+
+```
+sudo systemctl stop containerd
+```
+
+Next we need to edit the config file that manages all of containerd's settings.  
+This file is /etc/containerd/config.toml.  By default the file has an example
+configuration that is all commented out (please see all of the # signs). Let's 
+edit this file to include soci snapshotter as a plug in with the following:
+
+```
+proxy_plugins]
+  [proxy_plugins.soci]
+    type = "snapshot"
+    address = "/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock"
+```
+
+These lines tell containerd that there is a snapshot plugin named soci and to 
+communicate with it via a socket file.  Now let's restart containerd
+
+```
+sudo systemctl restart containerd
+```
+
+To confirm that containerd knows about soci run
+
+```
+ctr plugin ls
+```
+
+This will list all of the plugins confirm that there is a **soci** 
+plugin of type **io.containerd.snapshotter.v1**
+
+### Preparing soci snapshotter
+First we need to start the snapshotter binary, it lives in the `out` directory.
+In the example command we arbitrarily are pushing the logs to a file named
+`~/soci-snapshotter-logs`, but please feel free to choose a different location : 
+
+```
+sudo ./soci-snapshotter-grpc&> ~/soci-snapshotter-logs &
+```
+
+Once the snapshotter is running we can call the rpull command from the soci cli.
+This command reads the manfiest from the registry and mounts FUSE filesystems
+for each layer.   
+
+In this command the `--soci-index-digest` needs to be the digest of the soci
+index manifest.  The final argument is the tag of the image : 
+
+```
+sudo ./soci image rpull --plain-http --soci-index-digest sha256:f5f2a8558d0036c0a316638c5575607c01d1fa1588dbe56c6a5a7253e30ce107 localhost:5000/rabbitmq:latest
+```
+
+After running this command you will see a minimal output.  If you see all of the
+layers being fetched now, then soci has not worked.  It should look something
+like : 
+
+```
+using SOCI index digest: sha256:f5f2a8558d0036c0a316638c5575607c01d1fa1588dbe56c6a5a7253e30ce107
+fetching sha256:7e2a3a93... application/vnd.docker.distribution.manifest.v2+json
+fetching sha256:31b721ac... application/vnd.docker.container.image.v1+json
+```
+### Running ctr
+Now that all of the mounts are set up we can run the image using the following
+command in ctr.  We need to specify which snapshotter we shall use and we will
+use the `--net-host` binary flag.  Then we pass in the two main arguments, our
+image registry and the id of the container : 
+
+```
+sudo ctr run --snapshotter soci --net-host localhost:5000/rabbitmq:latest sociExample
+```
```

Comment:
Suggest a concluding sentence here. Not sure what. "Congratulations Lazy bones!" seems a little harsh.

Also, maybe we should have a couple of sentences here telling them what to look for in the system logs or `ctr` output or  `~/soci-snapshotter-logs` to verify that it did the thing, and to understand a little bit more about what it did.